src/Http/Controller/AdvsController.php

```diff
@@ -166,6 +223,12 @@ public function index($category = null, $city = null)
             }
             if (isset($param['cat'])) {
                 unset($param['cat']);
+                if (setting_value('visiosoft.module.advs::translatable_slug')){
+                    return redirect(fullLink(
+                        $param,
+                        route('adv_list_seo_mlang', [trans('visiosoft.module.advs::slug.category'),$category->slug])
+                    ));
+                }
```

Comment:
duplicated this return check multiple times, make it a function and call it instead of this.